libs/utils/decorate.js

```diff
@@ -54,9 +54,16 @@ export function decorateBlockText(el, config = ['m', 's', 'm'], type = null) {
   const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
   if (!el.classList.contains('default')) {
     if (headings) {
-      headings.forEach((h) => h.classList.add(`heading-${config[0]}`));
+      const lastHeading = headings[headings.length - 1];
+      if (type === 'allowDetailHeading') {
+        lastHeading?.classList.add(`heading-${config[0]}`);
+      } else {
+        headings.forEach((h) => h.classList.add(`heading-${config[0]}`));
+      }
```

Comment:
Maybe I'm missing something, but here's how I see this: before, all headings would get a `heading-{size}` class. With this change, a block using `allowDetailHeading` would just apply that class to the last heading, leaving all others without class (pun intended?). On the test pages, I noticed the xxxl headline is no longer styled as such:

| Before | After |
| ------------- | ------------- |
| <img width="544" alt="Screenshot 2024-09-04 at 12 40 02" src="https://github.com/user-attachments/assets/6573fccf-5ba8-49ce-97a6-9e2941cfc5e7"> |  <img width="544" alt="Screenshot 2024-09-04 at 12 40 14" src="https://github.com/user-attachments/assets/a11a3fde-9e3b-414c-af9c-4c8938907630"> |

I think the approach should be switched to "all headings except the first get the `heading-{size}` class" or whatever you deem correct to cover all the use-cases.

Reply:
Thanks for catching that. Makes more sense to switch the approach like you said and less type checking in the code. 👍